docs/source/tutorial/introduction.rst

```diff
@@ -0,0 +1,439 @@
+
+    G = rx.PyGraph()
+    G.extend_from_edge_list([(0, 1), (0, 2)])
+    new_node = G.add_node("spam")
+    print(rx.connected_components(G))
+    degrees = []
+    for node in G.node_indices():
+        degrees.append(G.degree(node))
+    print(degrees)
+
+.. jupyter-execute::
+
+    G.remove_node(new_node)
+    G.extend_from_edge_list([(0, 3), (0, 4), (1, 2)])
+    rx.transitivity(G)
+
+See the :ref:`algorithm_api` API documentation section for a list of example
+functions and how to use them.
+
+Drawing graphs
+==============
+
+There are two visualization functions provided in retworkx for visualizing
+graphs. First there is :func:`~retworkx.visualization.mpl_draw` which uses the
+`matplotlib <https://matplotlib.org/>`__ library to render the
+visualization of the graph. The :func:`~retworkx.visualization.mpl_draw`
+function relies on the :ref:`layout-functions` provided with retworkx to
+generate a layout (the coordinates to draw the nodes of the graph) for the
+graph (by default :func:`~retworkx.spring_layout` is used). For example:
+
+.. jupyter-execute::
+
+    import matplotlib.pyplot as plt
+
+    G = rx.generators.generalized_petersen_graph(5, 2)
+    subax1 = plt.subplot(121)
+    mpl_draw(G, with_labels=True, ax=subax1)
+    subax2 = plt.subplot(122)
+    layout = rx.shell_layout(G, nlist=[range(5, 10), range(5)])
+    mpl_draw(G, pos=layout, with_labels=True, ax=subax2)
+
+The second function is :func:`~retworkx.visualization.graphviz_draw` which
+uses `Graphviz <https://graphviz.org/>`__ to generate visualizations. For
+example:
+
+.. jupyter-execute::
+
+    from retworkx.visualization import graphviz_draw
+
+    G = rx.generators.heavy_hex_graph(7)
+    # set data payload to index
+    for node in G.node_indices():
+        G[node] = node
+
+    def node_attr_fn(node):
+        attr_dict = {
+            "style": "filled",
+            "shape": "circle",
+            "label": str(node)
+        }
+        # Data nodes are yellow
+        if node < 7 * 7:
+            attr_dict["color"] = "yellow"
+            attr_dict["fill_color"] = "yellow"
+        # Syndrome nodes are black
+        elif node >= 7 * 7 and node < (7 * 7) + ((7 - 1) * (7 + 1) / 2):
+            attr_dict["color"] = "black"
+            attr_dict["fill_color"] = "black"
+            attr_dict["fontcolor"] = "white"
+        # Flag quits are blue
+        else:
+            attr_dict["color"] = "blue"
+            attr_dict["fill_color"] = "blue"
+        return attr_dict
+
+    graphviz_draw(G, node_attr_fn=node_attr_fn, method="neato")
+
+Generally when deciding which visualization function to use there are a few
+considerations to make. :func:`~retworkx.visualization.mpl_draw` is a better
+choice for smaller graphs or cases where you want to integrate your graph
+drawing as part of a larger visualization.
+:func:`~retworkx.visualization.graphviz_draw` is typically a better choice
+for larger graphs because Graphviz is a dedicated tool for drawing graphs.
```

Comment:
```suggestion
Generally, when deciding which visualization function to use, there are a few
considerations to make. :func:`~retworkx.visualization.mpl_draw` is a better
choice for smaller graphs or cases where you want to integrate your graph
drawing as part of a larger visualization.
:func:`~retworkx.visualization.graphviz_draw` is typically a better choice
for larger graphs, because Graphviz is a dedicated tool for drawing graphs.
```